test/benchmarks/mongoBench/runner.js

```diff
@@ -9,33 +9,10 @@ function percentileIndex(percentile, total) {
   return Math.max(Math.floor((total * percentile) / 100 - 1), 0);
 }
 
-function timeDoneTask(task, ctx) {
-  return new Promise((resolve, reject) => {
-    let called = false;
-    const start = performance.now();
-    task.call(ctx, err => {
-      const end = performance.now(start);
-      if (called) return;
-      if (err) return reject(err);
-      return resolve((end - start) / 1000);
-    });
-  });
-}
-
 async function timeTask(task, ctx) {
-  // Some tasks are async, so they take callbacks.
-  if (task.length) {
-    return timeDoneTask(task, ctx);
-  }
-
   const start = performance.now();
-  const ret = task.call(ctx);
-  let end = performance.now();
-
-  if (ret && ret.then) {
-    await ret;
-    end = performance.now();
-  }
+  await task.call(ctx);
+  const end = performance.now();
```

Comment:
I would like to discuss this.  The bson benchmarks are synchronous.  I removed the logic to conditionally await the return of `task.call` because you can await synchronous expressions, but this does add one additional microtask before ending the timing to synchronous tasks.  I think this is okay - I think that the additional time taken before ending the timing is negligible compared to the task execution.

But I'm open to adding back the logic to only conditionally await the tasks.

Reply:
I think it's worth fixing, nodejs has `utils/type` module that exports a `isAsyncFunction` type checker we can use. Which means we must pass `async functions` to the `.task(` method to get them awaited, I think in the sync case I'd save the result of task.call(x) and if it is a promise throw, just to fail if someone introduces a manual promise retuning task

Reply:
From slack: Going with an .isAsync or .isSync flag of some kind